core/src/apps/cardano/sign_tx.py

```diff
@@ -4,32 +4,42 @@
 from trezor.crypto import base58, hashlib
 from trezor.crypto.curve import ed25519
 from trezor.messages.CardanoSignedTx import CardanoSignedTx
-from trezor.messages.CardanoTxAck import CardanoTxAck
-from trezor.messages.CardanoTxRequest import CardanoTxRequest
 
-from apps.cardano import CURVE, seed
+from apps.cardano import CURVE, MAINNET_PROTOCOL_MAGIC, TESTNET_PROTOCOL_MAGIC, seed
 from apps.cardano.address import (
     derive_address_and_node,
     is_safe_output_address,
     validate_full_path,
 )
-from apps.cardano.layout import confirm_sending, confirm_transaction, progress
+from apps.cardano.layout import confirm_sending, confirm_transaction
 from apps.common import cbor
 from apps.common.paths import validate_path
 from apps.common.seed import remove_ed25519_prefix
 
+if False:
+    from typing import List, Tuple
+    from trezor.messages.CardanoSignTx import CardanoSignTx
+    from trezor.messages.CardanoTxInputType import CardanoTxInputType
+    from trezor.messages.CardanoTxOutputType import CardanoTxOutputType
+
 # the maximum allowed change address.  this should be large enough for normal
 # use and still allow to quickly brute-force the correct bip32 path
 MAX_CHANGE_ADDRESS_INDEX = const(1000000)
-ACCOUNT_PREFIX_DEPTH = const(2)
+ACCOUNT_PREFIX_DEPTH = const(3)
+
+LOVELACE_INVALID = 45_000_000_000 * 1_000_000
```

Comment:
I think `LOVELACE_MAX_SUPPLY` would be more a accurate name

Reply:
Yeah, I was deciding between the two - both are used in Ledger code. I'll rename it.